bokeh/server/tornado.py

```diff
@@ -25,21 +25,46 @@
 from .application_context import ApplicationContext
 from .views.static_handler import StaticHandler
 
+
+def match_host(host, pattern):
+    """ Match host against pattern
+
+    >>> match_host('192.168.0.1', '192.168.0.1')
+    True
+    >>> match_host('192.168.0.1', '192.168.0.2')
+    False
+    >>> match_host('192.168.0.1', '192.168.*.*')
+    True
+    >>> match_host('alice', 'alice')
+    True
+    >>> match_host('alice', 'bob')
+    False
+    >>> match_host('alice', '*')
+    True
+    """
+    host = host.split('.')
+    pattern = pattern.split('.')
+
+    for h, p in zip(host, pattern):
+        if h == p or p == '*':
+            continue
+        else:
+            return False
+    return True
+
+
```

Comment:
Fixed, at least in this case.  I could easily be missing other similar issues.  I would be satisfied with the following simpler rule:

``python
def match_host(host, pattern):
    return host == pattern or pattern == '*'
```

Which, while less fancy, probably doesn't include any surprises.